death-causes-app/src/components/Question.tsx

```diff
@@ -0,0 +1,409 @@
+import React, { ChangeEvent, PureComponent, ReactElement } from "react";
+import "./QuestionMenu.css";
+import Button from "react-bootstrap/Button";
+import Popover from "react-bootstrap/Popover";
+import OverlayTrigger from "react-bootstrap/OverlayTrigger";
+import InputGroup from "react-bootstrap/InputGroup";
+import { Row, Col, Form } from "react-bootstrap";
+import "./Question.css";
+import MarkDown from "react-markdown";
+import { InputValidity } from "../models/Factors";
+import DropdownButton from "react-bootstrap/DropdownButton";
+import Dropdown from "react-bootstrap/Dropdown";
+import UnitPicker from "./UnitPicker";
+import { DropdownToggle } from "reactstrap";
+
+const BACKGROUNDCOLOR_DISABLED = "#c7c7c7";
+const TEXTCOLOR_DISABLED = "#999";
+const BACKGROUNDCOLOR_CHOICE = "#cef1f5";
+const ERROR_COLOR = "#fc0303";
+const WARNING_COLOR = "#bfa50d";
+
+interface I_Question<T> {
+  handleChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleIgnoreFactor: (factorname: string) => void;
+  name: string;
+  phrasing: string;
+  factorAnswer: T;
+  helpText: string;
+  featured: boolean;
+}
+
+interface I_QuestionStates {
+  ignore: boolean;
+}
+
+interface NumericQuestionProps extends I_Question<number> {
+  placeholder: string;
+  inputvalidity: InputValidity;
+  unitOptions: string[];
+  handleChangeUnit: (fname: string, newUnitName: string) => void;
+}
+
+interface StringQuestionProps extends I_Question<string> {
+  placeholder: string;
+  inputvalidity: InputValidity;
+  options: string[];
+}
+
+interface FormControlStyle {
+  [key: string]: string;
+}
+
+interface AbstractQuestionProps {
+  name: string;
+  ignore: boolean;
+  handleIgnoreBox: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  helpText: string;
+  phrasing: string;
+  secondLine: ReactElement | string;
+  featured: boolean;
+  unitText: string | React.ReactNode;
+}
+
+class Question extends React.PureComponent<AbstractQuestionProps> {
+  constructor(props: any) {
+    super(props);
+  }
+
+  helpBox() {
+    return (
+      <Popover id="popover-basic">
+        <Popover.Title as="h3">{this.props.name}</Popover.Title>
+        <Popover.Content>
+          <MarkDown>{this.props.helpText}</MarkDown>
+        </Popover.Content>
+      </Popover>
+    );
+  }
+
+  IgnoreCheckBox() {
+    return (
+      <div className="fixedWidth">
+        <Form.Check
+          onChange={this.props.handleIgnoreBox}
+          checked={this.props.ignore}
+          label="Ignore"
```

Comment:
hvad bruger vi dette label til?

Reply:
"Ignore" er den text der står ved siden af checkboxen, man kan trykke på for at ignorere spørgsmålet og gray-out'e tekstfeltet